### DISCUSSION TITLE
Async returns in extendType - Query

### DISCUSSION THREAD

**Discussion by MaximilianGaedke** [graphql-nexus/nexus#1163] [Q&A]
I have an async function in a for loop that I want to run first and then return data.
If I just return it normally, it simply returns the initial data, without the changes from the for loop or the function inside.
Does anyone have an idea where I could solve this?
Thank you very much for your help!

(+ I use Apollo GraphQL and Prisma)

**Comment by jferriday** [comment]
Do you have a code example of your resolver? It sounds like you'd need to map the return value of each async call to an array, then use Promise.all to return your data when these have resolved.

**Comment by Amaan215422** [comment]
GM mate
